packages/language-server/src/completion-items.ts

```diff
@@ -101,13 +118,28 @@ export function computeLSPKind(
   }
 }
 
-function createInsertText(suggestion: UI5XMLViewCompletion): string {
+function createTextEdit(
+  suggestion: UI5XMLViewCompletion,
+  originalPosition: Position
+): { textEdit: TextEdit; filterText: string } {
+  let position: SourcePosition | undefined = undefined;
   let insertText = suggestion.ui5Node.name;
+
+  // The filter text is used by VSCode/Theia to filter out suggestions that don't match the text the user wrote.
+  // Every character being replaced by the TextEdit (until the cursor position) should exist in the filter text.
+  // Since we replace the whole value (tag name/attribute key/attribute value) the filter text should contain
+  // the entire prefix (including the xml namespace in tag name, surrounding quotation marks for attribute value etc).
+  // In simple cases (like property name) this will be the name of the UI5 node.
+  let filterText = insertText;
   switch (suggestion.type) {
     // Attribute key
     case "UI5NamespacesInXMLAttributeKey": {
+      position = getXMLAttributeKeyPosition(suggestion.astNode);
+      insertText = `xmlns:${insertText}`;
+      // Namespace in xmlns attribute key should contain the "xmlns:" prefix
+      filterText = insertText;
+
       // Auto-insert the selected namespace
-      /* istanbul ignore else */
       if (suggestion.astNode.syntax.value === undefined) {
```

Comment:
out of scope for this PR but it looks like we need to model the "equals" in the ASTNode as well.

Reply:
Yes. It currently causes a bug when you have the `=` sign without a value (the `=` sign remains after accepting the proposal, which contains a value).